tensorboard/webapp/app_routing/route_config.ts

```diff
@@ -249,20 +313,24 @@ export class RouteConfigs {
 
       this.configMatchers.push(matcher);
 
-      if (matcher.config.routeKind !== null) {
-        this.routeKindToConfigMatchers.set(matcher.config.routeKind, matcher);
+      if (matcher.config.routeType === RouteType.CONCRETE) {
+        this.routeKindToConfigMatchers.set(
+          matcher.config.definition.routeKind,
+          matcher
+        );
 
-        if (matcher.config.defaultRoute) {
+        if (matcher.config.definition.defaultRoute) {
           this.defaultRouteConfig = matcher;
         }
       }
     }
   }
 
   private validateRouteConfigs(configs: RouteDef[]) {
-    const defaultRoutes = configs.filter((routeDef) => {
-      return Boolean(routeDef.routeKind !== null && routeDef.defaultRoute);
-    });
+    const concreteDefinitions = configs.filter((routeDef) => {
+      return Boolean(isConcreteRouteDef(routeDef));
```

Comment:
Can the `Boolean` be dropped, assuming TS knows `isConcreteRouteDef` returns a boolean?